auth/gcloud/aio/auth/session.py

```diff
@@ -33,8 +33,8 @@ def __init__(self, session: Optional[Session] = None, timeout: int = 10,
         self._timeout = timeout
 
     @abstractproperty
-    def session(self) -> Optional[Session]:
-        return self._session
+    def session(self) -> Session:
+        pass
 
```

Comment:
Is it ok to `pass` here? did you do it this way to let subclasses of `BaseSession` to override this method?

Reply:
@leanaha Great question. Since this is an `abstractproperty` (https://docs.python.org/3.7/library/abc.html#abc.abstractproperty), it _must_ be overridden by inheriting classes otherwise those classes cannot be instantiated:

> A class that has a metaclass derived from [ABCMeta](https://docs.python.org/3.7/library/abc.html#abc.ABCMeta) cannot be instantiated unless all of its abstract methods and properties are overridden.
https://docs.python.org/3.7/library/abc.html#abc.abstractmethod

You can still manually call an abstract method or property using the `super()` method but in this case you'll already have defined a `session` property anyway.